### DISCUSSION TITLE
Cannot stop MOOSE from renumbering nodeID (importing from .inp)

### DISCUSSION THREAD

**Discussion by Traiwit** [idaholab/moose#18358] [Q&A General] [answered]
Hi guys,

I'm having an issue with importing a mesh using `FileMesh`

I see it has this option: allow_renumbering=false, I tried but it still changed my nodeID

Here is an input file (below), as you can see the nodeID should jumps from 182 to 100001, but MOOSE continues with 183

![2](https://user-images.githubusercontent.com/79562439/125901237-eee6bd2d-e3da-46cb-a8f8-c96a792e15a3.PNG)

Is there a way to work around this?

```
[Mesh]
  type = FileMesh
  file = HTTRANS2020_G01HTv2_TETL_W2FF.inp
  allow_renumbering = false
  dim = 3
[]
```

Kind regards,
Traiwit

**Comment by GiudGiud** [comment, marked as answer]
Hello

I suspect the renumbering that is allowed/disallowed is the one by further mesh operations (refinement etc)
What do you need this for?

Guillaume

**Reply by Traiwit** [reply]
Hi @GiudGiud,

It will be very ideal for me if MOOSE can keep the node numbering to be exactly the same as the .inp file that I input, eg 182 then 100001, not 182 then 183.

I need it to be the same for Multi-point constraints (MPCs) which help me work with the fault within the domain.

For the small sample, I can use excel/python to work around this, but my work usually requires a couple million elements, and this will make it more difficult.

Traiwit

**Reply by GiudGiud** [reply]
Umm I cant see where this param is being used in the code. This might be a bug. 
Do you mind trying to add a `getMesh().allow_renumbering(false);` at the top of `void
FileMesh::buildMesh()` in FileMesh.C and see if it does the trick?

**Reply by GiudGiud** [reply]
(just a try, my suspicion is that this param isnt meant for that)

**Reply by Traiwit** [reply]
All good, thanks @GiudGiud 

I probs need python to work around this problem, thanks for trying though.

Traiwit

**Reply by GiudGiud** [reply]
Ok. I ll create an issue for this as we are trying to gather what our meshing needs are
